docs/programmers_guide/witness_formal_spec.md

```diff
@@ -124,23 +125,23 @@ Initial state of the stack is ` <empty> `;
 
 Witness: `HASH h2; BRANCH 0b11`
 
-Stack: `(hashNode(h1), h1)`
+Stack: `(HashNode{h1}, h1)`
 
 ---
 
 **2. Executing `HASH h2`**: push one more `hashNode` to the stack.
 
 Witness `BRANCH 0b11`
 
-Stack: `(hashNode(h2), h2); (hashNode(h1), h1)`
+Stack: `(HashNode{h2}, h2); (HashNode{h1}, h1)`
 
 ---
 
 **3. Executing `BRANCH 0b11`**: replace 2 items on the stack with a single `fullNode`.
 
 Witness: ` <empty> `
 
-Stack: `(fullNode{0: hashNode(h2), 1: hashNode(h1)}, KECCAK(h2+h1))`
+Stack: `(BranchNode{0: HashNode{h2}, 1: HashNode{h1}}, KECCAK(h2+h1))`
```

Comment:
Here is an example where having hashes as integers would be useful. Instead of defining "+" operation on type Byte[32], you can use integer arithmetics to define concatenation, like `(h2 mode 2^256)*2^256 + (h1 mode 2^256)` (and put the comment that it is actually concatenation)

Reply:
then it is better to just introduce a helper function `CONCAT_HASHES` with the implementation as you have suggested, that will make the spec more readable

Reply:
as it turns out, we already have a `CONCAT` helper function, so I'll just use that there